Couple of issues encountered after updating to v0.2.1

Hello, so far after upgrading to v0.2.1 we have ran into the following bugs:

---
Description: Creating a pool with null extra_specs crashes garm. 
Location: https://github.com/cloudbase/garm/blob/main/util/util.go#L154
Reason: null is unmarshalled into `data` and that causes `data["runner_install_template"]` to fail.
Logs:
```
panic: assignment to entry in nil map

goroutine 3042 [running]:
github.com/cloudbase/garm/util.MaybeAddWrapperToExtraSpecs({_, _}, {{0x3410d3c80000, 0x20}, {0x3410d3a27380, 0x7, 0x8}, {0x3410d3594820, 0x1e}, {0x3410d3a4a080, ...}, ...})
        /util/util.go:154 +0x3c7
github.com/cloudbase/garm/runner/pool.(*basePoolManager).addInstanceToProvider(_, {{0x3410d345d470, 0x24}, {0x0, 0x0}, {0x3410d3a11140, 0x10}, 0x2958, {0x3410d3c80000, 0x20}, ...})
        /runner/pool/pool.go:1111 +0xa8e
github.com/cloudbase/garm/runner/pool.(*basePoolManager).addPendingInstances.func1({{0x3410d345d470, 0x24}, {0x0, 0x0}, {0x3410d3a11140, 0x10}, 0x2958, {0x3410d3c80000, 0x20}, {0x3410d3ae6340, ...}, ...})
        /runner/pool/pool.go:1781 +0x1db
created by github.com/cloudbase/garm/runner/pool.(*basePoolManager).addPendingInstances in goroutine 1406
        /runner/pool/pool.go:1775 +0x66d
```
---
Description: Webhooks that dont match a pool return 500 server error
Location: https://github.com/cloudbase/garm/blob/main/runner/pool/pool.go#L273
Reason: fmt.Error should return a more sensible http code, maybe a 400 class response. E.g.
```
return fmt.Errorf("job %d should not be recorded: %w", jobParams.WorkflowJobID, runnerErrors.ErrNotFound)
```
Logs:
```
time=2026-06-15T22:41:45.981Z level=ERROR source=/runner/runner.go:896 msg="failed to handle workflow job" error="job 81541549816 should not be recorded"
```
---
Description: Pool manager found message is sent whether the pool manager is found or not.
Location: https://github.com/cloudbase/garm/blob/main/runner/runner.go#L873-L879
Reason: `slog.DebugContext(r.ctx, "found pool manager", "pool_manager", poolManager.ID())` should be moved to after err nil check on next line.

--- a/runner/pool/pool.go
+++ b/runner/pool/pool.go
@@ -270,7 +270,7 @@ func (r *basePoolManager) persistJobToDB(ctx context.Context, jobParams params.J
 			"is_new_job", isNewJob,
 			"job_action", jobParams.Action,
 		)
-		return fmt.Errorf("job %d should not be recorded", jobParams.WorkflowJobID)
+		return fmt.Errorf("job %d should not be recorded: %w", jobParams.WorkflowJobID, runnerErrors.ErrUnprocessable)
 	}
 
 	slog.DebugContext(

--- a/util/util.go
+++ b/util/util.go
@@ -139,6 +139,12 @@ func MaybeAddWrapperToExtraSpecs(ctx context.Context, param commonParams.Bootstr
 			return param
 		}
 	}
+	if data == nil {
+		// Unmarshal returned nil. An explicit `null` was probably passed.
+		// Any other valid json would either fail to unmarshal or unmarshal
+		// into an actual map we can use.
+		return param
+	}
 
 	if _, ok := data["runner_install_template"]; ok {
 		// User has already set a runner install template override. Do not touch.

